### PR TITLE
Upgraded HTTP to 0.5

### PR DESCRIPTION
**Pull request by olegmoz**
Upgraded HTTP to 0.5 with async Response

### CODE REVIEW

**Comment by paulodamaso** [comment]
~~@yegor256 I'm available for review, please assign me~~ Sorry wrong recipient

**Comment by paulodamaso** [comment]
@g4s8 I'm available for review, please assign me

**Comment by 0crat** [comment]
This pull request [#8](https://github.com/artipie/vertx-server/issues/8) is assigned to @hdouss[/z](https://www.0crat.com/u/hdouss), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/4354786b-d5f6-4f2d-b58d-b300e1181941); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/vertx/VertxSliceServer.java [hunk 1]
```diff
@@ -127,18 +128,24 @@ public void close() {
...
                     response.setChunked(true);
+                    final CompletableFuture<Void> promise = new CompletableFuture<>();
                     Flowable.fromPublisher(body).map(
                         buf -> {
                             final byte[] bytes = new byte[buf.remaining()];
                             buf.get(bytes);
                             return Buffer.buffer(bytes);
-                        }).subscribe(response.toSubscriber());
+                        })
+                        .doOnComplete(() -> promise.complete(null))
```
Comment (HDouss) [inline src/main/java/com/artipie/vertx/VertxSliceServer.java]: @olegmoz I would suggest to hold some data in the promise, maybe the body content, or the HttpServerResponse altogether. That will give some information to the promise handler, and will avoid for us to have `null` values.

File: src/test/java/com/artipie/vertx/VertxSliceServerTest.java [hunk 2]
```diff
@@ -79,14 +77,11 @@ public void basicGetRequest() throws IOException {
...
             vertx,
-            (line, headers, body) -> connection -> {
-                final int okay = 200;
-                connection.accept(
-                    okay,
-                    Collections.emptyList(),
-                    Flowable.fromArray(ByteBuffer.wrap(expected.getBytes()))
-                );
-            },
+            (line, headers, body) -> connection -> connection.accept(
```
Comment (HDouss) [inline src/test/java/com/artipie/vertx/VertxSliceServerTest.java:80]: @olegmoz Isn't there a way to test the promise that results from the Connection (or the Response.send)?

**Review by HDouss — CHANGES_REQUESTED** [review]
@olegmoz Please see 2 comments

File: src/test/java/com/artipie/vertx/VertxSliceServerTest.java (hunk 2 above)
Comment (olegmoz) [reply src/test/java/com/artipie/vertx/VertxSliceServerTest.java:80]: @HDouss I don't think there is a way and that is really needed. VertxSliceServerTest tests VertxSliceServer and it does not expose future produced by Slice. Here we are testing that server   actually sends response that Slice sent to connection. We also need to handle errors in Slice, but I will add it later in other PR. This PR is just to adapt VertxSliceServer  to updated HTTP module.

File: src/main/java/com/artipie/vertx/VertxSliceServer.java (hunk 1 above)
Comment (olegmoz) [reply src/main/java/com/artipie/vertx/VertxSliceServer.java]: @HDouss got rid of null

**Comment by olegmoz** [comment]
@HDouss made changes, please review

**Review by HDouss — APPROVED** [review]
@olegmoz Thanks.

**Comment by HDouss** [comment]
@g4s8 Please review and/or merge.

**Review by g4s8 — APPROVED** [review]

**Comment by 0crat** [comment]
Job audit: `ARC` didn't merge or close PR, performer @hdouss[/z](https://www.0crat.com/u/hdouss)[/z](https://www.0crat.com/u/hdouss) didn't make CR comments

**Comment by 0crat** [comment]
Job was finished in 19 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Pull request merged**